modules/network/src/main/java/org/apache/ignite/internal/network/DefaultMessagingService.java

```diff
@@ -674,15 +693,15 @@ protected void onStoppingInitiated() {
      * @param recipientNode Target cluster node.
      */
     private @Nullable InetSocketAddress resolveRecipientAddress(ClusterNode recipientNode) {
-        NetworkAddress recipientAddress = recipientNode.address();
-
         // Node name is {@code null} if the node has not been added to the topology.
         if (recipientNode.name() != null) {
-            return connectionManager.consistentId().equals(recipientNode.name()) ? null : createResolved(recipientAddress);
+            return connectionManager.consistentId().equals(recipientNode.name()) ? null : getFromCacheOrCreateResolved(recipientNode);
```

Comment:
Should we do the same on line 714?

Reply:
I think not, the node is not in the topology.
I thought about it, we may not delete this address later and it will hang in the air.